test/tests/misc/boundary_variable_check/tests

```diff
@@ -0,0 +1,20 @@
+[Tests]
+  issues = '#9734'
+  design = 'AuxKernels/index.md UserObjects/index.md'
+  [error]
+    requirement = 'The system shall error if a boundary restricted object depends on a variable that is not defined on any portion of the boundary. These boundary restricted objects include'
+    [aux]
+      type = RunException
+      input = test.i
+      expect_err = "'bad' of type 'SelfAux' depends on variable 'v'. However, that variable does not appear to have a superset of that object's block/boundary restriction."
+      detail = 'auxiliary kernels'
+    []
+    [uo]
+      type = RunException
+      input = test.i
+      expect_err = "'bad' of type 'NodalExtremeValue' depends on variable 'v'. However, that variable does not appear to have a superset of that object's block/boundary restriction."
+      cli_args = "AuxKernels/active='' Postprocessors/active='bad'"
+      detail = 'nodal user objects'
+    []
```

Comment:
what new requirement/detail would that add? Actually that would lead to no errors because this integrity check is for nodeset based objects and because the variables are all well defined on the nodeset along the interface, there would be no errors

Reply:
well you would have to restrict the variables as well.
I m thinking of distinguishing between internal sideset checks and external (to the mesh) checks in the test

Reply:
If you only have two domains, there's no way to restrict a Lagrange variable such that it's not defined on the interface nodeset

Reply:
yeah then you d need 3 domains

Reply:
So you want me to add a test for what kind of object(s) on an internal boundary?